src/converter/one/proto/Proto.java

```diff
@@ -140,4 +166,14 @@ private void ensureCapacity(int length) {
             buf = Arrays.copyOf(buf, newLength < 0 ? 0x7ffffff0 : Math.max(newLength, pos + length));
         }
     }
+
+    public static final class Mark {
```

Comment:
As in C++, mark can be just `long`. In Java, this makes even more sense, since JVM capabilities of allocation elimination are very limited.

Reply:
1daeb9af3f5e77b126d0f2328f4d7ae6bab238ae